### PR TITLE
feat: report a problem panels (OFF data + French SignalConso)

### PR DESCRIPTION
**Pull request by stephanegigandet**
Display a panel to offer to correct OFF data, + a SignalConso panel to report a product in France.
Can be extended later to add customer service contact info + NutriPatrol

image

### CODE REVIEW

**Comment by teolemon** [comment]
would be awesome to:
add UTM on SignalConso link
fire a Matomo event on our side (that way we have a proxy of most problematic products inside Matomo)

File: templates/api/knowledge-panels/report_problem/fr_report_product_signalconso.tt.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+Direction générale de la concurrence, de la consommation et de la répression des fraudes (DGCCRF).
+                `
+            },
+        },
+        {
+            "element_type": "image",
+            "image_element": {
+                "url": "[% static_subdomain %]/images/panels/report_problem/signalconso-button.png",
+                "alt": "Je fais un signalement sur SignalConso",
+                "link_url": "https://signal.conso.gouv.fr/fr/alimentaire-codebarres/faire-un-signalement?gtin=[% product.code %]"
```
Comment (teolemon) [inline templates/api/knowledge-panels/report_problem/fr_report_product_signalconso.tt.json:30]:
```suggestion
                "link_url": "https://signal.conso.gouv.fr/fr/alimentaire-codebarres/faire-un-signalement?gtin=[% product.code %]?utm_source=off&utf_medium=mobile&utm_campaign=product_page_kp"
```

Comment (teolemon) [reply templates/api/knowledge-panels/report_problem/fr_report_product_signalconso.tt.json:30]: removing &utf_medium=mobile

File: templates/api/knowledge-panels/report_problem/incomplete_or_incorrect_data.tt.json [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        "subtitle": "[% edq(lang('incomplete_or_incorrect_data_subtitle_off')) %]",
+        "icon_url": "[% static_subdomain %]/images/logos/off-logo-icon-light.svg",
+    },
+    "elements": [
+        {
+            "element_type": "text",
+            "text_element": {
+                "html": `
+                [% edq(lang('incomplete_or_incorrect_data_content_correct')) %]
+                [% edq(lang('incomplete_or_incorrect_data_content_correct_off')) %]
```
Comment (alexgarel) [inline templates/api/knowledge-panels/report_problem/incomplete_or_incorrect_data.tt.json:19]: edq here is not necessary.
We should instead escape backtick !

File: templates/api/knowledge-panels/root.tt.json [hunk 3]
```diff
@@ -14,6 +14,12 @@
                 "panel_id": "environment_card",
             },
         },
+        {
+            "element_type": "panel",
+            "panel_element": {
+                "panel_id": "report_problem_card",
+            },
+        },
```
Comment (alexgarel) [inline templates/api/knowledge-panels/root.tt.json:22]: minor: You put a if in the knowledge panel construction method, but not here !

**Review by alexgarel — APPROVED** [review]
Cool.

File: templates/api/knowledge-panels/report_problem/fr_report_product_signalconso.tt.json (hunk 1 above)
Comment (stephanegigandet) [reply templates/api/knowledge-panels/report_problem/fr_report_product_signalconso.tt.json]: Added.

File: templates/api/knowledge-panels/root.tt.json (hunk 3 above)
Comment (stephanegigandet) [reply templates/api/knowledge-panels/root.tt.json:22]: the if is for the signalconso panel, the report_problem_card is always there.

File: templates/api/knowledge-panels/report_problem/incomplete_or_incorrect_data.tt.json (hunk 2 above)
Comment (stephanegigandet) [reply templates/api/knowledge-panels/report_problem/incomplete_or_incorrect_data.tt.json:19]: right, thanks

**Comment by stephanegigandet** [comment]
Added a "Report a problem" tab at the top:

image

**Comment by teolemon** [comment]
Contribution tab link is going to empty when you are not logged in (I think it's meant to jump on the Data Quality panel)

**Comment by teolemon** [comment]
We could misuse the '''Source''' element as a way to add a clickable URL on mobile

**Pull request merged**